python/interpret-core/interpret/glassbox/_ebm/_ebm.py

```diff
@@ -2884,7 +2461,186 @@ def predict(self, X, init_score=None):
         return inv_link(scores, self.link_, self.link_param_)
 
 
-class DPExplainableBoostingClassifier(EBMModel, ClassifierMixin, ExplainerMixin):
+@dataclass(repr=False)
+class DPEBMModel(EBMModel):
+
+    # BEGIN: overwriting default values
+    # Preprocessor
+    max_bins: int = 32
+    max_interaction_bins: None = None
+
+    # Stages
+    interactions: Literal[0] = 0
+    # Ensemble
+    validation_size: Union[
+        Annotated[int, Field(strict=True, ge=0)],
+        Annotated[float, Field(ge=0.0, lt=1.0)],
+    ] = 0
+    outer_bags: Annotated[int, Field(ge=1)] = 1
+    inner_bags: Literal[0] = 0
+    # Boosting
+    greediness: Literal[0] = 0
+    cyclic_progress: Literal[1.0] = 1.0
+    smoothing_rounds: Literal[0] = 0
+    interaction_smoothing_rounds: Literal[0] = 0
+    max_rounds: NonNegativeInt = 300
+    early_stopping_rounds: Literal[0] = 0
+    early_stopping_tolerance: Literal[0] = 0
+    # Trees
+    min_samples_leaf: Literal[0] = 0
+    min_hessian: Literal[0] = 0
+
+    # Overall
+    random_state: Optional[int] = None
```

Comment:
I see my earlier comment above about having some of these attributes not exist is handled here by deleting these attributes essentially.  It still feels to me that having a separate non-DP class is a more expected way to handle this, and the subtractive approach I think is also more prone to errors if for example there is an accidental mismatch in one of the attribute names.

Thoughts?